smartcontractkit/external-adapters-js
# Feature/37419 update withmetrics middleware

## Closes #37419

## Description
Updated withMetrics middleware to record new error types to prometheus.
......

## Steps to Test

1. `yarn test:unit`
2. `yarn test:integration`
3. Run an adapter locally with metrics enabled & check the `/metrics` endpoint

## Quality Assurance
